.github/workflows/builder_go_slsa3.yml

```diff
@@ -37,10 +37,19 @@ on:
         required: true
         type: string
       upload-assets:
-        description: "If true provenance is uploaded to a GitHub release for new tags."
+        description: >
```

Comment:
@olivekl any feedback on the change in the file?
We're adding a new option called `upload-url` in this PR.
`upload-assets` is the option we already had, and it indicates that the workflow should automatically upload the provenance / binaries on new tags / releases.

`upload-url` is a new option for users to pass an upload url, because they sometimes create the release themselves in their workflow. In this case the workflow does not run on new release, it runs on a trgger event like `workflow_dispatch` (manual trigger) and the workflow creates the release before handing over the execution to our sSLSA generator.

/cc @asraa